projects/packages/waf/src/class-waf-runner.php

```diff
@@ -260,8 +276,12 @@ public static function get_rules_from_api() {
 			throw new \Exception( 'Site is not registered' );
 		}
 
-		$response = Client::wpcom_json_api_request_as_user(
-			sprintf( '/sites/%s/waf-rules', $blog_id )
+		$response = Client::wpcom_json_api_request_as_blog(
```

Comment:
Actually, now that I think about it, I think we can make this work with a blog check (by pulling the list of upgrades for that blog, rather than the user). But might be worth asking crew if there's an easy way to do this request as the main connected Jetpack user?

Reply:
Going to leave as it is for now then, we can figure out a way to change in another version if we decide that working with blog connections is too cumbersome.

Reply:
As per @miguelxpn:
> The cron job doesn't work with a connected user so we have to connect as a blog, you can sandbox your api and make the permission check return true and it will work, I'll try to write a patch for the permission check to grab the main connected user of that particular blog to see if they're an automattician now that we'll be testing it more internally